server/protocols/webadmin/webadmin-mailbox/src/test/java/org/apache/james/webadmin/routes/EventDeadLettersRoutesTest.java

```diff
@@ -327,12 +344,460 @@ void deleteShouldRemoveEvent() {
             deadLetters.store(new EventBusTestFixture.GroupA(), EVENT_1).block();
 
             with()
-                .delete("/events/deadLetter/groups/" + SERIALIZED_GROUP_A + "/events/" + UUID_1);
+                .delete("/events/deadLetter/groups/" + SERIALIZED_GROUP_A + "/" + UUID_1);
 
             when()
-                .get("/events/deadLetter/groups/" + SERIALIZED_GROUP_A + "/events/" + UUID_1)
+                .get("/events/deadLetter/groups/" + SERIALIZED_GROUP_A + "/" + UUID_1)
             .then()
                 .statusCode(HttpStatus.NOT_FOUND_404);
         }
     }
+
+    @Nested
+    class RedeliverAllEvents {
+        private Group groupA;
+        private Group groupB;
+
+        @BeforeEach
+        void nestedBeforeEach() {
+            EventCollector eventCollectorA = new EventCollector();
+            EventCollector eventCollectorB = new EventCollector();
+            groupA = new EventBusTestFixture.GroupA();
+            groupB = new EventBusTestFixture.GroupB();
+            eventBus.register(eventCollectorA, groupA);
+            eventBus.register(eventCollectorB, groupB);
+        }
+
+        @Test
+        void postRedeliverAllEventsShouldCreateATask() {
+            deadLetters.store(groupA, EVENT_1).block();
+
+            given()
+                .queryParam("action", EVENTS_ACTION)
+            .when()
+                .post("/events/deadLetter")
+            .then()
+                .statusCode(HttpStatus.CREATED_201)
+                .header("Location", is(notNullValue()))
+                .body("taskId", is(notNullValue()));
+        }
+
+        @Test
+        void postRedeliverAllEventsShouldHaveSuccessfulCompletedTask() {
+            deadLetters.store(groupA, EVENT_1).block();
+
+            String taskId = with()
+                .queryParam("action", EVENTS_ACTION)
+                .post("/events/deadLetter")
+                .jsonPath()
+                .get("taskId");
+
+            given()
+                .basePath(TasksRoutes.BASE)
+            .when()
+                .get(taskId + "/await")
+            .then()
+                .body("status", is("completed"))
+                .body("taskId", is(taskId))
+                .body("additionalInformation.successfulRedeliveriesCount", is(1))
+                .body("additionalInformation.failedRedeliveriesCount", is(0))
+                .body("type", is(EventDeadLettersRedeliverTask.TYPE))
+                .body("startedDate", is(notNullValue()))
+                .body("submitDate", is(notNullValue()))
+                .body("completedDate", is(notNullValue()));
+        }
+
+        @Test
+        void postRedeliverAllEventsShouldRedeliverAndRemoveEventFromDeadLetters() {
```

Comment:
You are not asserting the redeliver, just the *remove* pârt. Please adapt the test name.